### PR TITLE
Fix #24710 - Update countries list

### LINKED ISSUE #24710: Update countries list / CLDR to version 36
### Preconditions (*)
1. Magento 2.3.2 & 2.3-develop

### Steps to reproduce (*)
1. Go to stores -> configuration -> General -> General -> Countries list.

### Expected result (*)
1. All existing countries available.

### Actual result (*)
1. Missing countries. For example Curacao, Sint Maarten are missing and there are also 3 Municipalities (Bonaire, Sint Eustasius and Saba) which form the Caribbean Netherlands which is also missing. I have read that Kosovo and Canary Islands are also missing.
2. I looked into Magento's ZendFramework1 library that is currently using version 24 of the (Common Locale Data Repository) which is currently already at version 36.

We need to update the Locale data and fix the entries in the directory_country* tables.

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Description (*)

This is fix for #24710
Note, that Canary Islands weren't added as it is part of Spain and doesn't contain it's own iso_3 code

### Fixed Issues (if relevant)

1. magento/magento2#24710: Update countries list / CLDR to version 36

### Manual testing scenarios (*)

1. Go to Stores -> Configuration -> General -> General -> Country Options
2. Check if **Allow Countries** contain following countries: Caribbean Netherlands, Curacao, Kosovo, Sint Maarten.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Directory/Setup/Patch/Data/AddMissingCountries.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
+<?php
+/**
```
Comment (osrecio) [inline app/code/Magento/Directory/Setup/Patch/Data/AddMissingCountries.php]: Can you rename the filename to be more specific, like the countries are you adding. [Take a look](https://github.com/magento/magento2/tree/2.3-develop/app/code/Magento/Directory/Setup/Patch/Data).

**Review by osrecio — CHANGES_REQUESTED** [review]

File: app/code/Magento/Directory/Setup/Patch/Data/AddMissingCountries.php (hunk 1 above)
Comment (Bartlomiejsz) [reply app/code/Magento/Directory/Setup/Patch/Data/AddMissingCountries.php]: Hi @osrecio, done

**Comment by osrecio** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @osrecio. Thank you for your request. I'm working on Magento instance for you

**Review by osrecio — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @osrecio, here is your [new Magento instance](https://pr-24927.instances.magento-community.engineering).
Admin access: [https://pr-24927.instances.magento-community.engineering/admin](https://pr-24927.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by magento-engcom-team** [comment]
Hi @osrecio, thank you for the review.
[ENGCOM-6043](https://jira.corp.magento.com/browse/ENGCOM-6043) has been created to process this Pull Request
:eight_spoked_asterisk: @osrecio, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by osrecio** [comment]
I tested in dev environment and local environment and the countries are displayed correctly, and you can choose them.

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**